src/main/java/sg/ncl/MainController.java

```diff
@@ -5116,4 +5131,182 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
     private String getExperimentMessage(String expName, String teamName) {
         return "Experiment " + expName + " in team " + teamName;
     }
+
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.GET)
+    public String addMember(@PathVariable String teamId, Model model) {
+        model.addAttribute("addMemberForm", new AddMemberForm());
+        return "add_member";
+    }
+
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.POST)
+    public String addMember(@PathVariable String teamId, @Valid AddMemberForm addMemberForm,
+                            final RedirectAttributes redirectAttributes)  throws WebServiceRuntimeException {
+        log.info("Starting to adding members by email for team {}", teamId);
+
+        String emails[] = addMemberForm.getEmails().split("\\r?\\n");
+
+        JSONArray jsonArray = new JSONArray();
+        for (int i = 0; i< emails.length; i++){
+            if (!VALID_EMAIL_ADDRESS_REGEX.matcher(emails[i]).matches()) {
+                addMemberForm.setErrMsg(EMAIL_ADDRESS_IS_NOT_VALID);
+                redirectAttributes.addFlashAttribute(MESSAGE, EMAIL_ADDRESS_IS_NOT_VALID);
+                return REDIRECT_ADD_MEMBER;
+            }
+            jsonArray.put(emails[i]);
+        }
+
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("emails", jsonArray.toString());
+
+        HttpEntity<String> request = createHttpEntityWithBody(jsonObject.toString());
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity responseEntity = null;
+        try {
+            responseEntity = restTemplate.exchange(properties.addMemberByEmail(teamId), HttpMethod.POST, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for adding members by email: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return REDIRECT_ADD_MEMBER;
+        }
+
+        String responseBody = responseEntity.getBody().toString();
+
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error;
+            try {
+                error = objectMapper.readValue(responseBody, MyErrorResource.class);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)